Normalized spacing/brackets in pseudocode

pjotrek-b opened the pull request:
In order to make the pseudocode less susceptible to misinterpretation, I've added {} even for one-liners for all "if" and "for".

For the sake of consistency, I also normalized the spacing of "if", "for" and "while" calls as follows:
  * Space between command (if, for, while) and arguments
  * Space between operators.
  * No space after/before closing (). Unless inside calls a function, to avoid "))" for better readability.

Although these changes are mostly cosmetic, I think it's good to have consistent and well-readable code syntax throughout a document.

pjotrek-b commented:
I've also added additional typo corrections.

JeromeMartinez commented on ffv1.md:
```diff
@@ -704,17 +708,20 @@ log2_run[41]={                                                |
                                                               |
 if (run_count == 0 && run_mode == 1) {                        |
-    if (get_bits(1)) {                                        |
+    if ( get_bits(1) ) {                                      |
```
spaces near parenthesis are added here as you indicated ("Space between command (if, for, while) and arguments") but I don't see the change for other parts ("if (x + run_count <= w)", "if (run_index)"... below and lot of others), or did I miss something?

pjotrek-b replied:
It seems inconsistent, but I followed a rule:
  * If the statement in brackets ends with a bracket, I've kept/added the space, so it's clearer to read.
  * If it does not end in brackets, there's no need for space for readability.

JeromeMartinez replied:
Ho, sorry, right you wrote it.
On my side I think it looks weird, I would prefer one of them only (all with space or none with space), but nothing big and we can keep that as is. @michaelni any preference?

michaelni replied:
the "mixed" style looks a bit odd to me too. Not that it really matters, its just spaces ...

pjotrek-b replied:
I actually took the "))"-avoiding from existing code-styles in the paper. Thought it was on purpose that it was already mixed therefore.

Anyways: I can normalize all to non-space. No problem.

Shall I?

pjotrek-b replied:
Okay, I've now checked and seen that it's actually just this single if() with extra spaces. Thought there were more cases like that... :see_no_evil:

So I've now changed this so everything should be conform now :smile:

JeromeMartinez replied:
I may have missed something but I see that you removed space from get_ur_golomb() part but not the one I am commenting (below if (run_count == 0 && run_mode == 1)).
PS: small stuff, just spaces, but as I find nothing else... :).

michaelni commented:
the spaces within while() seems inconsistent

pjotrek-b commented:
I applied the same "no ))" rule to while.
I've now also removed the spaces there :smile:

btw: I've changed `!byte_aligned()`to `!byte_aligned( )`, since all empty parentheses in the rest of the document have a space between.

I hope I didn't miss anything anymore?

pjotrek-b commented:
OMG! :scream:
Now I know why I couldn't find " ( " anymore!
It's not spaces (0x20) in the document, but something else (0xC2A0)...

![ffv1_specs_space-0xc2a0](https://user-images.githubusercontent.com/2419195/53482215-06273080-3a7f-11e9-8b99-4a4c79a291c3.png)

pjotrek-b commented:
I've never seen this before, and I'll check if this was done by some process on my side...

JeromeMartinez commented:
> It's not spaces (0x20) in the document, but something else (0xC2A0)...

it is the UTF-8 code for [non-breaking space](https://en.wikipedia.org/wiki/Non-breaking_space).
Not bad in the idea, more complex in the writing, maybe could be replaced by classic spaces.

pjotrek-b commented:
Oh.
Me (and my VIM) were absolutely unaware of this, so I've very likely fudged regular spaces over some of them.

I've now edited my vimrc to show NBSPs.

pjotrek-b commented:
I can go through the document again and use NBSPs accordingly.

JeromeMartinez commented:
@michaelni any formatting preference about NBSP, i.e. should we keep them?
On my side I prefer removing them for simplicity but not against having them.
then we stop with this space formatting stuff :).

pjotrek-b commented:
btw: I just saw that in not all pseudo-code blocks in the initial ffv1.md use NBSPs, but regular space.

And sorry for opening this can of worms, but who could have known? :cry:

michaelni commented:
> @michaelni any formatting preference about NBSP, i.e. should we keep them?

No oppinion :)

pjotrek-b commented:
I think plain spaces will probably suffice in this case.
Since the width of the pseudo-code blocks is limited to <80 characters anyways, I expect it to be less likely to be broken in different renderings.

I'll replace it with spaces for the time being.

pjotrek-b commented:
Why does noone tell me about the whitespace landmines set across this document, before you let me walk in, looking like a bull in a china shop? :cry: :wink:

Therefore I kept is as separate commit.
Can you cherry pick?

michaelni commented:
I see commits with identical and duplicated commit messages (537b947d05dfe461451ed69059c6c0d513c098af, a9c64342a0e04448bcfa16c79624e727fcaf7aeb ). That obviously proofs that these commit messages are insufficiently describing the change, because they work for more than one change.
also i can cherry pick approved changes but it may be quicker and avoid mistakes (on my side) if pull requests do not need to be edited. Also if cherry picking is wanted refering to a commit by its hash is more foolproof than describing it
Anyway ill cherry pick them in order upto the first where i see an issue

michaelni commented:
622bab9ecb013f265a343c286e8b7230015cbb1b adds spaces around if ( get_bits(1) ) { 
2b9c6fab0a8962f53c271b48403f14fab6a5e62d removes the spaces around if ( get_bits(1) ) { 

Making a change in a commit that is undone in the same pull request in a later commit is ugly and makes the changes a mess and hard to follow

pjotrek-b commented:
I hope I got it fixed now? :sweat_smile: 

I've rebased it, reworded duplicate commit messages and removed "undo" commits by applying the final change in the first commit already (then skipped in rebase).

retokromer commented:
@pjotrek-b I guess there is still [`get_sr_golomb(k)`](https://github.com/FFmpeg/FFV1/blob/master/ffv1.md#signed-golomb-rice-codes) to be unified.

pjotrek-b commented:
I'd leave that one as it is.

Normalizing that one seemed bloated to me.
It's clear to read, and every code has its justified 1-or-2-liner-ifs :wink:

michaelni commented:
44b780418d276726246e569a2d396f07eb21db31 is a merge, this does not belong in a pull request

pjotrek-b commented:
Sorry :(
Don't know how that merge happened.

I'm not very experienced at all with git pull requests or branching/merging...

I think the easiest would be for me to start over and leave this messed up pull request behind :wink:

pjotrek-b commented:
I think/hope I've removed the merge now.

I'll try to rebase it to resolve the conflict without merging. Wish me luck! :sweat_smile:
(Thanks @JeromeMartinez)

pjotrek-b commented:
Should be fine now.
Please check :smile:

michaelni commented:
generated pdf is filled with gibberish. like ".# Abstract" or "category = "std" litterally in the document

dericed commented:
@michaelni I've been unable to reproduce the gibberish. Also his PR only impacts ffv1.md and not the frontmatter files or the Makefile so I'm not sure how you're getting a gibberish pdf from this branch.

michaelni commented:
@dericed, i cannot reproduce it either now. I assume this was PEBCAK on my side, ill take another look tomorrow and if i dont spot anything then ill apply

dericed commented:
nudge :)
